src/opm/io/eclipse/EGrid.cpp

```diff
@@ -28,44 +30,203 @@
 #include <string>
 #include <sstream>
 
+#include <math.h>
+
 namespace Opm { namespace EclIO {
 
-EGrid::EGrid(const std::string &filename) : EclFile(filename)
+using NNCentry = std::tuple<int, int, int, int, int,int, float>;
+
+EGrid::EGrid(const std::string &filename, std::string grid_name) :
+    EclFile(filename), inputFileName { filename }, m_grid_name {grid_name}
 {
-   EclFile file(filename);
+    initFileName = inputFileName.parent_path() / inputFileName.stem();
+
+    if (this->formattedInput())
+        initFileName += ".FINIT";
+    else
+        initFileName += ".INIT";
+
+    std::string lgrname = "global";
 
-   auto gridhead = get<int>("GRIDHEAD");
+    m_nncs_loaded = false;
+    actnum_array_index = -1;
+    nnc1_array_index = -1;
+    nnc2_array_index = -1;
 
-   nijk[0] = gridhead[1];
-   nijk[1] = gridhead[2];
-   nijk[2] = gridhead[3];
+    int hostnum_index = -1;
+
+    for (size_t n = 0; n < array_name.size(); n++) {
+
+        if (array_name[n] == "ENDLGR")
+            lgrname = "global";
+
+        if (array_name[n] == "LGR") {
+            auto lgr = this->get<std::string>(n);
+            lgrname = lgr[0];
+            lgr_names.push_back(lgr[0]);
+        }
+
+        if (array_name[n] == "NNCHEAD"){
+            auto nnchead = this->get<int>(n);
+
+            if (nnchead[1] == 0)
+               lgrname = "global";
+            else
+               lgrname = lgr_names[nnchead[1] - 1];
+        }
+
+        if (lgrname == grid_name) {
+            if (array_name[n] == "GRIDHEAD") {
+                auto gridhead = get<int>(n);
+                nijk[0] = gridhead[1];
+                nijk[1] = gridhead[2];
+                nijk[2] = gridhead[3];
+
+                m_radial = gridhead[26] > 0 ? true: false;
```

Comment:
Please note that this (accessing `gridhead[26]`)  combined with the unit test's [definition](https://github.com/OPM/opm-common/blob/4c48f9bcf91f91974c6cf505b5f4129814d2e589/tests/test_EclRegressionTest.cpp#L315)
```C++
std::vector<int> gridhead = {1,2,3,2,0,0};
```
(`gridhead.size() == 6`) generates a guaranteed out-of-bounds indexing error.

I tried to fix this locally in the obvious way&mdash;extending the size of `gridhead` in `test_EclRegressionTest.cpp:gridCompare`&mdash;but that only led to more unit test failures.

What is the correct way to fix the out-of-bounds error here?

Reply:
Thanks @bska  for spotting this error/bug. I think that the best way to fix this is to initialize the variable m_radial to false in addition to adding a check for size of gridhead

```c++
if (gridhead.size() > 26)
    m_radial = gridhead[26] > 0 ? true: false;
```

I can add a PR to fix this if you agree with this solution